<!-- -*- mode: XML -*- -->
<launch>
    <arg name="connect_on_startup"   default="true"/>
    <arg name="calibrate_on_startup" default="true"/>
    <arg name="engage_on_startup"    default="true"/>

    <arg name="odom_frame"        default="odom"/>
    <arg name="base_frame"        default="wheel_odom_link"/>
    <arg name="publish_odom"        default="true"/>
    <arg name="publish_odom_tf"     default="false"/>

    <arg name="publish_current"        default="true"/>
    
    <arg name="output"   default="screen"/>
    <arg name="respawn"   default="false"/>
    
    <node pkg="differential_odrive" type="odrive_node" name="odrive" ns="odrive" 
            output="$(arg output)" respawn="$(arg respawn)">

        <param name="wheel_track" value="0.77"/>
        <param name="tyre_circumference" value="0.03852"/>

        
        <param name ="dual_odrive" value="true"/>
        <param name ="left_sn" type="string" value="206E35925748"/>
        <param name ="right_sn" type="string" value="205938915848"/>

        <param name ="flip_left_direction" value="true"/>
        <param name ="flip_right_direction" value="false"/>

        <param name="odom_frame"   value="$(arg odom_frame)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        
        <param name="connect_on_startup"   value="$(arg connect_on_startup)"/>
        <param name="calibrate_on_startup" value="$(arg calibrate_on_startup)"/>
        <param name="engage_on_startup"    value="$(arg engage_on_startup)"/>
        
        <param name="publish_odom"         value="$(arg publish_odom)"/>
        <param name="publish_odom_tf"      value="$(arg publish_odom_tf)"/>

        <param name="publish_current"      value="$(arg publish_current)"/>
    </node>
</launch>
